x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/indexlifecycle/UnfollowFollowIndexStep.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.core.indexlifecycle;
+
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.action.admin.indices.close.CloseIndexRequest;
+import org.elasticsearch.action.admin.indices.open.OpenIndexRequest;
+import org.elasticsearch.client.Client;
+import org.elasticsearch.cluster.ClusterState;
+import org.elasticsearch.cluster.metadata.IndexMetaData;
+import org.elasticsearch.xpack.core.ccr.action.PauseFollowAction;
+import org.elasticsearch.xpack.core.ccr.action.UnfollowAction;
+
+import java.util.Map;
+
+final class UnfollowFollowIndexStep extends AsyncActionStep {
+
+    static final String NAME = "unfollow-index";
+
+    UnfollowFollowIndexStep(StepKey key, StepKey nextStepKey, Client client) {
+        super(key, nextStepKey, client);
+    }
+
+    @Override
+    public void performAction(IndexMetaData indexMetaData, ClusterState currentClusterState, Listener listener) {
+        String followerIndex = indexMetaData.getIndex().getName();
+        Map<String, String> customIndexMetadata = indexMetaData.getCustomData("ccr");
+        if (customIndexMetadata == null) {
+            listener.onResponse(true);
+            return;
+        }
+
+        pauseFollowerIndex(followerIndex, listener);
+    }
+
+    void pauseFollowerIndex(final String followerIndex, final Listener listener) {
```

Comment:
One other thing: I'm not sure whether these are worth breaking out into their own AsyncActionSteps. That seems more in line with the intent of the model and make it clearer which part had a problem, but it might be worth it if it makes the interval we have to close the index for shorter. I'll investigate this.

Reply:
I see, it is good to know at what part of the unfollow index step the policy is (pause index following, close index, unfollow or open index). Is there way to do that without adding a step for each action?